experiment/src/org/labkey/experiment/ExpDataIterators.java

```diff
@@ -3055,11 +3003,85 @@ private void writeRowsToFile(TypeData typeData)
             if (typeData.dataRows.isEmpty())
                 return;
 
+            // for cross type import, write to further partitions
+            if (_isCrossFolderUpdate)
+            {
+                ExpObject dataType = typeData.dataType;
+
+                // query for container
+                Map<String, List<Integer>> containerRows = new HashMap<>();
+
+                TableInfo tableInfo = null;
+                SimpleFilter filter = null;
+
+                if (_isSamples)
+                {
+                    filter = new SimpleFilter(FieldKey.fromParts("MaterialSourceId"), dataType.getRowId());
+                    filter.addCondition(FieldKey.fromParts("Name"), typeData.dataIds, CompareType.IN);
+                    tableInfo = ExperimentService.get().getTinfoMaterial();
+                }
+                else
+                {
+                    filter = new SimpleFilter(FieldKey.fromParts("ClassId"), dataType.getRowId());
+                    filter.addCondition(FieldKey.fromParts("Name"), typeData.dataIds, CompareType.IN);
+                    tableInfo = ExperimentService.get().getTinfoData();
+                }
+
+                Map<String, Object>[] rows = new TableSelector(tableInfo, Set.of("name", "container"), filter, null).getMapArray();
+                for (Map<String, Object> row : rows)
+                {
+                    String name = (String) row.get("name");
+                    String dataContainer = (String) row.get("container");
+                    int sampleRowId = typeData.dataIds.indexOf(name);
+                    containerRows.computeIfAbsent(dataContainer, k -> new ArrayList<>()).add(sampleRowId);
+                }
+
+                for (String containerId : containerRows.keySet())
+                {
+                    Container container = _containerMap.get(containerId);
+                    if (container == null)
+                    {
+                        Container folder = ContainerManager.getForId(containerId);
+                        _context.getErrors().addRowError(new ValidationException("You don't have the required permission to update " + (_isSamples ? "samples" : "data") + " in the container: " + (folder != null ? folder.getName() : containerId)));
```

Comment:
```suggestion
                        _context.getErrors().addRowError(new ValidationException("You don't have the required permission to update " + (_isSamples ? "samples" : "data") + " in the folder: " + (folder != null ? folder.getName() : containerId)));
```
(This is based on a comment made in a recent bug bash. App users don't know much about containers.)